zilliqa/src/exec.rs

```diff
@@ -1036,6 +1068,7 @@ impl PendingState {
         PendingState {
             pre_state: state,
             new_state: HashMap::new(),
+            gas_left: Gas::new(0),
```

Comment:
Where is this used? I tried to find it in the precompile implementation but couldn't. I'm slightly confused as to what this is actually for?